compiler/next/include/chpl/parsing/parsing-queries.h

```diff
@@ -69,36 +69,46 @@ namespace parsing {
    */
   const uast::Builder::Result& parseFile(Context* context, UniqueString path);
 
-  using LocationsMap = std::unordered_map<ID, Location>;
-  /**
-   This query returns a map from ID to Location for the uAST parsed from a
-   particular file.
-   */
-  const LocationsMap& fileLocations(Context* context, UniqueString path);
 
   // These functions can't return the Location for a Comment
   // because Comments don't have IDs. If Locations for Comments are needed,
-  // instead use the locations field from the result of parseFile.
+  // instead use the astToLocation field from the result of parseFile.
 
   /**
    This query returns the Location where a particular ID appeared.
    It cannot be used for Comments because Comments don't have IDs set.
    If Locations for Comments are needed, use the locations field from
    the result of parseFile.
    */
-  const Location& locateID(Context* context, ID id);
+  const Location& locateId(Context* context, ID id);
   /**
-   This function just runs locateID on ast->id(). Similarly to locateID,
+   This function just runs locateId on ast->id(). Similarly to locateID,
    it cannot be used to get a Location for a Comment.
    */
-  const Location& locate(Context* context, const uast::ASTNode* ast);
+  const Location& locateAst(Context* context, const uast::ASTNode* ast);
 
   using ModuleVec = std::vector<const uast::Module*>;
   /**
    This query returns a vector of parsed modules given a file path.
    */
   const ModuleVec& parse(Context* context, UniqueString path);
 
+  /**
+   This query parses a toplevel module by name. Returns nullptr
+   if no such toplevel module can be found.
+   */
+  const uast::Module* getToplevelModule(Context* context, UniqueString name);
```

Comment:
Is this covered by unit tests? I see idToAst and idToParentId but I don't see this one.

Reply:
I'll add a test for it, thanks.